In [2]:
# Question 1. import numpy, pandas, and stats models
import pandas as pd
import numpy as np
import statsmodels.api as sm

In [7]:
# Question  2. Load the data into a pandas DataFrame. Calculate summary statistics and make sure everything looks sensible.
df = pd.read_csv("wellness_data.csv")
df.describe()

wellness        age50     age37_49      age1836  diabetes_pre  \
count  2151.000000  2151.000000  2151.000000  2151.000000   2151.000000   
mean      0.606230     0.320316     0.363552     0.316132      4.927940   
std       0.488699     0.466707     0.481134     0.465073     21.650112   
min       0.000000     0.000000     0.000000     0.000000      0.000000   
25%       0.000000     0.000000     0.000000     0.000000      0.000000   
50%       1.000000     0.000000     0.000000     0.000000      0.000000   
75%       1.000000     1.000000     1.000000     1.000000      0.000000   
max       1.000000     1.000000     1.000000     1.000000    100.000000   

          spend_pre   spend_year1  
count   2151.000000   2151.000000  
mean     468.418498    613.322585  
std     1042.161526   1198.288793  
min        0.000000      0.000000  
25%       39.692305     87.197165  
50%      155.285400    256.259100  
75%      453.463450    632.707450  
max    19941.530000  16630.750000

Question 2 comments:

In [18]:
# Question 3. Measure associaiton between X and Y variable
# a. Calculate average of spend_year1 among people with wellness==1 and wellness==0

# Average of spend_year1 when wellness == 1
avg_wellness_1 = df[df['wellness'] == 1]['spend_year1'].mean()
print("Year 1 spending average when wellness = 1: ", avg_wellness_1)

# Average of spend_year1 when wellness == 0
avg_wellness_0 = df[df['wellness'] == 0]['spend_year1'].mean()
print("Year 1 spending average when wellness = 0: ", avg_wellness_0)

# b. Estimate a regression of spend_year1 on wellness.


# Estimate a regression of spend_year1 on wellness.
y = df['spend_year1']
X = df['wellness']
X = sm.add_constant(X)
model = sm.OLS(y, X).fit()
print(model.summary())

Year 1 spending average when wellness = 1:  587.1873023282209
Year 1 spending average when wellness = 0:  653.5591939662337
                            OLS Regression Results                            
Dep. Variable:            spend_year1   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     1.576
Date:                Thu, 10 Oct 2024   Prob (F-statistic):              0.210
Time:                        19:46:14   Log-Likelihood:                -18299.
No. Observations:                2151   AIC:                         3.660e+04
Df Residuals:                    2149   BIC:                         3.661e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
-------

Question 3 comments:
Year 1 spending average when wellness = 1:  587.19
Year 1 spending average when wellness = 0: 653.56

Because the Wellness Variable has a coefficient of -66.37, that means that enrolling in the wellness program is associated with a decrease in year1 spending off $66.37. The constant of 653.56 is exactl what we calculated for the average year 1 spending when wellness = 0



**Question 4. Consider selection bias. Imagine and report  any sources of systematic differences between people who participate in the wellness program and people who do not.**

- People who are already active and healthy may be more likely to opt into the program
- Socioeconomic status may play a role in whether or not people opt in to the program. This can be things like an individuals income or educational backround

**Question  5. Identify confounds**

**a. Differences in health spending**
1. Pre Existing conditions
2. Weight
3. Age
4. Disability

**b. Use of wellness programs**
1. health consiousness
2. health conditions
3. Age
4. Employer incentive

**c. note confounds**


**Question 6. Which confounds are measured?**
 - Age
 - Diabetes (pre-existing condition)

In [24]:
# Question 7. Investgiate whether diabetes_pre is a confound, and adjust for it
# a. Find the fraction of people with diabetes_pre  who use the wellness program,
#     and the fraction of people without diabtes_pre  who use the wellness program.

frac_with_diabetes = df[df['diabetes_pre'] == 100]['wellness'].mean()
frac_without_diabetes = df[df['diabetes_pre'] == 0]['wellness'].mean()
print("Fraction of people with diabetes_pre who use the wellness program: ", frac_with_diabetes)
print("Fraction of people without diabetes_pre who use the wellness program: ", frac_without_diabetes)

# b. Find average spending among people with diabetes_pre, and among people without.

avg_spending_with_diabetes = df[df['diabetes_pre'] == 100]['spend_year1'].mean()
avg_spending_without_diabetes = df[df['diabetes_pre'] == 0]['spend_year1'].mean()
print("Average spending among people with diabetes_pre: ", avg_spending_with_diabetes)
print("Average spending among people without diabetes_pre: ", avg_spending_without_diabetes)

# C. Measure the association between wellness and spend_year1 among people with
#   diabetes_pre (as in step 3) and, separately, without diabetes_pre.


# d. To adjust for diabetes_pre, estimate a regression of spend_year1 on
#    wellness and diabetes_pre. What is the coefficient on wellness now?

y = df['spend_year1']
X = df[['diabetes_pre', 'wellness']]
X = sm.add_constant(X)
model = sm.OLS(y, X).fit()
print(model.summary())


Fraction of people with diabetes_pre who use the wellness program:  0.49056603773584906
Fraction of people without diabetes_pre who use the wellness program:  0.6122249388753056
Average spending among people with diabetes_pre:  1359.7051591509437
Average spending among people without diabetes_pre:  574.6347836945722
                            OLS Regression Results                            
Dep. Variable:            spend_year1   R-squared:                       0.020
Model:                            OLS   Adj. R-squared:                  0.020
Method:                 Least Squares   F-statistic:                     22.48
Date:                Thu, 10 Oct 2024   Prob (F-statistic):           2.19e-10
Time:                        20:10:20   Log-Likelihood:                -18277.
No. Observations:                2151   AIC:                         3.656e+04
Df Residuals:                    2148   BIC:                         3.658e+04
Df Model:                           2             

Question 7 comments/findings:

- Fraction of people with diabetes_pre who use the wellness program:  0.49
- Fraction of people without diabetes_pre who use the wellness program:  0.61
- Average spending among people with diabetes_pre: 1359.71
- Average spending among people without diabetes_pre: 574.63

The average spending for people with diabetes is considerabily higher than the average spending for those without. Also, People with diabetes are less likely to opt into the wellness program.

- coef on wellness of -47.77 means people who do not join the wellness program spend 47.77 dollars more given all else constant.
- people with diabeties have 7.79 dollars more spending

In [27]:
df.columns

Index(['wellness', 'age50', 'age37_49', 'age1836', 'diabetes_pre', 'spend_pre',
       'spend_year1'],
      dtype='object')

In [28]:
# Question 8. Adjust for all potential confounds in the data
# Specifically, estimate a regression of spend_year1 on wellness,
# and also spend_pre, the age variables, and diabetes_pre.
# What is the coefficient on wellness?

y = df['spend_year1']
X = df[['wellness', 'age50', 'age37_49', 'age1836', 'diabetes_pre', 'spend_pre']]
X = sm.add_constant(X)
model = sm.OLS(y, X).fit()
print(model.summary())



                            OLS Regression Results                            
Dep. Variable:            spend_year1   R-squared:                       0.303
Model:                            OLS   Adj. R-squared:                  0.302
Method:                 Least Squares   F-statistic:                     186.9
Date:                Thu, 10 Oct 2024   Prob (F-statistic):          1.56e-165
Time:                        20:23:20   Log-Likelihood:                -17910.
No. Observations:                2151   AIC:                         3.583e+04
Df Residuals:                    2145   BIC:                         3.587e+04
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const          233.6331     27.433      8.516   

Question 8 comments:

What is the coefficient on wellness?
 - The coefficient on wellness is 13.585. Considering all other variables as contant, people who join the wellness program are associated with an increase of 13.58 dollars in year 1 spending.

**Question 9: Did it matter that we adjusted for measured confounds? Were there any unmeasured confounds?**

-Yes it did matter that we adjusted for the measured confound because it completely changed the coefficients of the variables. This showed a correlation between year 1 spending and the opting in of the wellness program. There may still be unmeasured confounds out there that we are not accounting for. For example, I think one that is unmeasured could be an employer incentive for opting into the wellness program.